src/Tests/ImageBundleTest.php

```diff
@@ -0,0 +1,117 @@
+   *
+   * @var \Drupal\media_entity\MediaBundleInterface
+   */
+  protected $bundleStorage;
+
+  /**
+   * The test user.
+   *
+   * @var \Drupal\User\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->container->get('module_installer')->install(['image_bundle_test']);
+    $entityManager = $this->container->get('entity.manager');
+    $this->bundleStorage = $entityManager->getStorage('media_bundle');
+    $this->testBundle = $this->bundleStorage->load('images_media_bundle');
+
+    $this->adminUser = $this->drupalCreateUser([
+        'view media',
+        'create media',
+        'update media',
+        'update any media',
+        'delete media',
+        'delete any media',
+        'access media overview',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests image media bundle creation from config files.
+   */
+  public function testMediaBundleCreationFromModule() {
+    $type_configuration = array(
+      'source_field' => 'field_image',
+      'gather_exif' => false
+    );
+
+    $this->assertTrue((bool) $this->testBundle, 'The media bundle from default configuration has been created in the database.');
+    $this->assertEqual($this->testBundle->get('label'), 'Images media bundle', 'Correct label detected.');
+    $this->assertEqual($this->testBundle->get('description'), 'Media bundle for Images', 'Correct description detected.');
+    $this->assertEqual($this->testBundle->get('type'), 'image', 'Correct plugin ID detected.');
+    $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+    $this->assertEqual($this->testBundle->get('field_map'), [], 'Correct field map detected.');
+  }
+
+  /**
+   * Tests item creation and thumbnail
+   */
+  public function testMediaBundleItemCreation(){
+    // Define the media item name.
+    $name = $this->randomMachineName();
+
+    $image_files = $this->drupalGetTestFiles('image');
+    $testImage = File::create((array) current($image_files));
+    $file_path = $this->container->get('file_system')->realpath($testImage->getFileUri());
+    $edit = [
+      'name[0][value]' => $name,
+      'files[field_image_0]' => $file_path,
+    ];
+
+    // Save it and verify the title.
+    $this->drupalPostForm('media/add/' . $this->testBundle->id(), $edit, t('Save and publish'));
+    $this->assertTitle($edit['name[0][value]'] . ' | Drupal');
+    $media_id = \Drupal::entityQuery('media')->execute();
+    $media_id = reset($media_id);
+    $edit['id'] = $media_id;
+
+    // Go to media item list.
+    $this->drupalGet('admin/content/media');
+    $this->assertResponse(200);
+    $this->assertText($name);
+
+    // Check if the thumbnail exists or not.
+    $xpath = "//table[@class='views-table views-view-table cols-7']/tbody/tr[1]/td[1]/img";
+    $this->assertFieldByXPath($xpath, NULL, "Thumbnail for the media exists.");
```

Comment:
Check for the actual file name. This assert will also pass with the placeholder thumbnail.